### PR TITLE
MariaDB support

### PR DESCRIPTION
**Pull request by Sybrand**
# Added MariaDB support

- Added MariaDB plugin and Dockerfile.
- Added MariaDB to documentation.
- Fixed typos in backup-deploy.json.

This plugin should also work with rhl mysql image, but is untested.

### CODE REVIEW

**Review by WadeBarnes — COMMENTED** [review]
I've had an initial look.  Testing required.

**Comment by esune** [comment]
I will take a look and test  asap, and report back. We have a MariaDB database, I will setup a test instance to conduct tests safely.

**Comment by esune** [comment]
Tested by creating a backup and restoring it on MariaDB: everything works great. I would expect `mysql` to be the same since they run basically the same engine, right?

Thanks @Sybrand for the plugin :)

**Comment by Sybrand** [comment]
@esune the plugin should work with mysql, yes. One may have to use a different source image for the dockerfile - I used registry.access.redhat.com/rhscl/mariadb-102-rhel7 (depreceted b.t.w!) because it matches what we're using for our mariadb.

**Review by esune — APPROVED** [review]

**Comment by WadeBarnes** [comment]
@Sybrand, Can you try using a newer base image.  If it's anything like PostgreSQL as long as the backup container is running a version that is the same or newer than the database you're backing up or restoring things work just fine.

**Comment by esune** [comment]
> 
> 
> Can you try using a newer base image
@WadeBarnes Is it worth using images from https://registry.fedoraproject.org so that we can be on more recent versions than the ones in the Red Hat Developer Portal?

**Comment by esune** [comment]
Alternatively: `registry.access.redhat.com/rhscl/mariadb-102-rhel7`. I am not sure whether these images can be pulled from outside OCP, so that was where my question about using the Fedora-based images was going, in addition to those being usually updated sooner.

**Comment by WadeBarnes** [comment]
I'm fine with Fedora images.  As long as it works, it does not really make much difference.

**Comment by Sybrand** [comment]
@WadeBarnes the most recent version I see on redhat is mariadb-103-rhel7 ( https://catalog.redhat.com/software/containers/rhscl/mariadb-103-rhel7/5c9927a3bed8bd28d0e931d0) - but when I tried to use that, I couldn't - some credential error.
@esune I'm not seeing any straight up MariaDB image on https://registry.fedoraproject.org .
I have no issue upping the version - but I don't have a clue where to go for an appropriate image.

**Comment by esune** [comment]
>
> @esune I'm not seeing any straight up MariaDB image on https://registry.fedoraproject.org .
> I have no issue upping the version - but I don't have a clue where to go for an appropriate image.

Their website is not great, this is the latest one that I can see:  registry.fedoraproject.org/f31/mariadb - would have to look up what version that corresponds to.

The issue about being able to pull is what I was referring to above, unless you're on a machine with a `rhscl` license it will nt let you pull the image.

**Comment by Sybrand** [comment]
> > @esune I'm not seeing any straight up MariaDB image on https://registry.fedoraproject.org .
> > I have no issue upping the version - but I don't have a clue where to go for an appropriate image.
> 
> Their website is not great, this is the latest one that I can see: registry.fedoraproject.org/f31/mariadb - would have to look up what version that corresponds to.
> 
> The issue about being able to pull is what I was referring to above, unless you're on a machine with a `rhscl` license it will nt let you pull the image.

Cool - got it https://registry.fedoraproject.org/repo/f31/mariadb/tags/ - looks to be 10.3

File: docker/Dockerfile_MariaDB [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from registry.fedoraproject.org/f31/mariadb
```
Comment (Sybrand) [inline docker/Dockerfile_MariaDB:1]: @esune I've changed it to fedora - tested backup and verification, works for me.

**Review by esune — APPROVED** [review]
Looks good to me, I am happy with the Fedora images - will let @WadeBarnes confirm this works for him as well.

**Comment by WadeBarnes** [comment]
Thanks @Sybrand, I'll have a look tomorrow.

**Review by WadeBarnes — APPROVED** [review]

**Pull request merged**